File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    static int s = 1, e = 0;
    public int solution(int n) {
        int ret = 0;
        int[] a = new int[n];
        // int k = 10;
        int sum = 0;
        while(e != n){
            // System.out.println(s + " :: " + e);
            if(sum < n) sum += s++;
            else if(sum > n) sum -= e++;
            if(sum == n) { ret++; sum += s++;}
        }
        return ret;
    }
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    s = e = 1
    ret = 0
    total = 1
    while s <= n:
        if total == n:
            ret += 1
            total -= s
            s += 1
        elif total < n:
            e += 1
            total += e
        else:
            total -= s
            s += 1
            
    return ret
        

"""
1 = 1
2 = 2
3 = 1 + 2, 3
4 = 4
5 = 2 + 3, 5
6 = 1 + 2 + 3, 6
7 = 3 + 4, 7
8 = 8


"""


